java-vtl-script/src/main/java/no/ssb/vtl/script/operations/FilterOperation.java

```diff
@@ -66,9 +67,8 @@ protected DataStructure computeDataStructure() {
         DataPointBindings dataPointBindings = new DataPointBindings(componentBindings, getDataStructure());
         return stream.map(dataPointBindings::setDataPoint)
                 .filter(bindings -> {
-                    VTLBoolean resolved = (VTLBoolean) predicate.resolve(dataPointBindings);
-                    Boolean predicate = resolved.get();
-                    return predicate == null ? false : predicate;
+                    VTLObject resolved = predicate.resolve(dataPointBindings);
+                    return resolved.get() == null ? false : VTLBoolean.of((Boolean) resolved.get()).get();
```

Comment:
This is the same behaviour as before. If `predicate` is not of type VTLBoolean, a `ClassCastException` will be thrown. This should be handled earlier, for instance with a type check in the constructor. I will create a separate issue for it.

Reply:
As discussed with @takvamborgen we will create a separate issue for this.

Reply:
Indeed, that is the reason we pass around the types in wrapper (VTLTyped) so that we can check when needed.

In filter operation, the only predicate we should get is a VTLExpression that returns VTLBoolean when calling getVTLType(). 

https://github.com/statisticsnorway/java-vtl/blob/9dafa429a4f0ae27cb9dec5005a3b7e2e4ee44fa/java-vtl-script/src/main/java/no/ssb/vtl/script/visitors/ExpressionVisitor.java#L199-L201

Reply:
Yeah it would be nice to make sure that the expression the filter uses is checked using the same mechanism. 

IMO the type check does not belong in the FilterOperation. It is better to change predicate to VTLExpression<VTLBoolean> and see where the compiler complains.